stdlib/weakref.pyi

```diff
@@ -119,7 +125,13 @@ class WeakKeyDictionary(MutableMapping[_KT, _VT]):
     @overload
     def pop(self, key: _KT, default: _VT | _T = ...) -> _VT | _T: ...
     if sys.version_info >= (3, 9):
+        @overload
+        def __or__(self, other: Mapping[_KT, _VT]) -> Self: ...
+        @overload
         def __or__(self, other: Mapping[_T1, _T2]) -> WeakKeyDictionary[_KT | _T1, _VT | _T2]: ...
+        @overload
+        def __ror__(self, other: Mapping[_KT, _VT]) -> Self: ...
+        @overload
```

Comment:
Adding the new overloads here is incorrect: `weakref.WeakKeyDictionary.__or__` always returns `WeakKeyDictionary` instances, even in a subclass:

```pycon
>>> from weakref import WeakKeyDictionary
>>> class Foo(WeakKeyDictionary): ...
...
>>> type(Foo() | Foo())
<class 'weakref.WeakKeyDictionary'>
```

We should leave `WeakKeyDictionary` as it is. The same goes for `WeakValueDictionary`.

Reply:
Ah, you are right. It calls copy, but it's copy does not return Self.